### PR TITLE
Feature/registration tests

### PR DESCRIPTION
**Pull request by nghamilton**
Indentation tidy for Registration.hs

### CODE REVIEW

File: consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs [hunk 1]
```diff
@@ -647,277 +686,309 @@ data SoftwareStatement = SoftwareStatement {
...
-  ( AsError e
-  , MonadError e m
-  , FromJSON a
-  ) => Value -> m a
-fromVal = rToM . fromJSON where
- rToM = \case
-         Error s -> throwError . (_ParseError #) $ s
-         Success a -> pure a
+fromVal ::
+     forall e m a. (AsError e, MonadError e m, FromJSON a)
```
Comment (ajmcmiddlin) [inline consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs]: Prefer previous.

File: consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs [hunk 2]
```diff
@@ -647,277 +686,309 @@ data SoftwareStatement = SoftwareStatement {
...
 maybeErrors :: (AsError e, MonadError e m) => e -> Maybe a -> m a
 maybeErrors e = maybe (throwError e) pure
 
-aesonClaimsToMetaData :: forall e m.
-  ( AsError e
-  , MonadError e m
-  ) =>
- AesonClaims -> m ClientMetaData
+aesonClaimsToMetaData ::
+     forall e m. (AsError e, MonadError e m)
```
Comment (ajmcmiddlin) [inline consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs]: Prefer previous.

File: consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs [hunk 3]
```diff
@@ -647,277 +686,309 @@ data SoftwareStatement = SoftwareStatement {
...
+jwtToRegoReq audPred issPred jwk jwt
     -- TODO: seperate predicates here for the JWT?
-    validationSettings = defaultJWTValidationSettings audPred & issuerPredicate .~ issPred
-    c2m c = c ^. unregisteredClaims . to aesonClaimsToMetaData
-    jwsHead = undefined
+ = do
+  let validationSettings =
+        defaultJWTValidationSettings audPred & issuerPredicate .~ issPred
+      c2m c = c ^. unregisteredClaims . to aesonClaimsToMetaData
+      jwsHead = undefined
```
[the remark this answers is not in the document]
Comment (nghamilton) [reply consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs]: I figured `error "not yet yet done"` and undefined would convey similar enough meaning, but I'm happy to change that approach. Or probably most likely I'll include changes that actually implement it in the next PR.

[the remark this answers is not in the document]
Comment (ajmcmiddlin) [reply consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs]: All good. I'll sometimes use `undefined`s while checking other things are compiling, so wasn't sure if this was just forgotten or a TODO.

File: consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs [hunk 4]
```diff
@@ -182,11 +181,17 @@ _FapiGrantTypes :: Prism' GrantTypes FapiGrantTypes
 _FapiGrantTypes = prism'
   (\(FapiGrantTypes f) -> f)
   (\(GrantTypes grantTypes) ->
-    if grantTypes `isSubsetOf` permittedGrantTypes && requiredGrantTypes `isSubsetOf` grantTypes then Just . FapiGrantTypes . GrantTypes $ grantTypes else Nothing
+    if grantTypes
```
[the remark this answers is not in the document]
Comment (nghamilton) [reply consumer-data-au-api-types/src/Web/ConsumerData/Au/Api/Types/Auth/Registration.hs]: This is probably going to a matter of manually fixing some of these up afterwards I reckon. If you're happy with the rest of the changes, I'll merge this PR. I'm very fine if you or anyone else modifies any code I produce to beautify it further in the future. (And I won't set anything up to automatically beatify, as I am guessing brittany can't be guaranteed to not do styling silliness).

**Pull request merged**